### ISSUE TITLE
Add comma separating thousands to number fields and fix text/string types

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-reporting#478]
For all number fields, add a comma to separate thousands for better readability. 

<img width="664" alt="Image" src="https://github.com/user-attachments/assets/e6479216-df85-4c09-916c-2be2e9c6a6d2" />

Based on [BC Gov writing guidance](https://www2.gov.bc.ca/gov/content/governments/services-for-government/service-experience-digital-delivery/web-content-development-guides/web-style-guide/writing-guide/numbers) numbers should have commas separating

Notes: 

- ~~**CRA business number** and **partner company CRA business number** fields in administration are both numbers, so make sure implementing the separators doesn't affect those fields.~~ CRA business number is being turned into a string as part of https://github.com/bcgov/cas-registration/issues/3379
- commas are being used in some fields in reporting. we need to ensure this is consistently applied across the app. 
- don't apply this to the BC Energy Regulator Well Authorization Number(s)
- don't apply this to the BC Corporate Registry Number

### Dev Checklist:
- [ ] review all number input fields in Admin, Reg, & Reporting to check that the comma-separated-digits widget is being used where it should be
- [ ]~~stop comma-separating values in CRA business number fields~~

Commas where there shouldn't be:
- CRA: https://github.com/bcgov/cas-registration/issues/3379
- well authorization: https://github.com/bcgov/cas-reporting/issues/827

Missing commas (appears to be anywhere that's got type "string" in the rjsf schema):
- allocation of emissions:

![Image](https://github.com/user-attachments/assets/d64a28a5-fb33-4cc2-a5ea-ae028d81e5d5)

- compliance summary readonly fields: 

![Image](https://github.com/user-attachments/assets/e4625e2d-2a6e-42dc-b146-d5ecabba1429)
- final review readonly fields: 

![Image](https://github.com/user-attachments/assets/e7e8e1f8-a84f-4dd9-ba52-48f9599fd018)

![Image](https://github.com/user-attachments/assets/40e4899d-fc7c-489b-a7a0-f29a281feef5)

**Comment by zoeyli-46** [comment]
@patriciarussellCAS would this be a post-MVP priority? It seems like a nice to have but not a must have. But this would be standardizing all of the number fields to have commas and not just on the allocation of emissions page

**Comment by patriciarussellCAS** [comment]
Definitely a nice to have, but likely pretty easy to do.

**Comment by patriciarussellCAS** [comment]
Note: I have been seeing commas in some of the forms, but it is inconsistent. For example, on activity pages it works for emissions, but won't display for the equivalent emission calculated. 
Also, the commas are not being displayed on the final review page. Maybe need to refine this ticket to capture the

**Comment by patriciarussellCAS** [comment]
Tagging @zoeyli-46  and @nanyangpro

**Comment by patrickisaac** [comment]
Refinement Notes:
- @zoeyli-46 / @nanyangpro to check guidance around this situation

**Comment by zoeyli-46** [comment]
@patriciarussellCAS [BC Gov writing guidance](https://www2.gov.bc.ca/gov/content/governments/services-for-government/service-experience-digital-delivery/web-content-development-guides/web-style-guide/writing-guide/numbers) indicates numbers should have commas separating instead of spaces, 

<img width="497" alt="Image" src="https://github.com/user-attachments/assets/b44dc8ab-6946-4b8a-94a2-358b9dfe0ed4" />

**Comment by andrea-williams** [comment]
@gdalcengio IIRC you implemented the comma-separated thousands to some number fields - where did you implement that logic?

**Comment by gdalcengio** [comment]
@andrea-williams there's no CAS logic for it, it is included it the NumberField component

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EVgJJj4hhDlPoWyg9iA14-cBiMze1DPxkDxlLQx4QQ0Ulw?e=0qjjHr).

I can verify that:
- [x] In reporting:
   - [x] When value > 999, it is comma-separated, tested in the following pages:
      - [x] Activity forms
      - [x] Emission Summary page
      - [x] Production Data page
      - [x] Allocation of Emissions page
      - [x] Additional Reporting Data page
      - [ ] New Entrant Information page
      - [x] Operation Emission Summary
      - [x] Compliance Summary page
      - [x] Final Review page
   - [x] When value <= 999, it is comma-separated, tested in all pages above
- [x] In administration:
   - [x] The `CRA Business Number` field is not comma-separated
   - [x] The `BCGHG ID` fields in both the operation and facility view is not comma-separated
   - [x] The `Telephone Number` in the contacts page is not comma-separated both in create mode and view mode

A few things I noticed:
- The `Equivalent Value` is not set to 4 decimal digits anymore, same with values in the Compliance Summary page, Emissions Summary page, Production Data page, and Additional Reporting Data page
- In the Allocation of Emissions page, the text fields are not aligned anymore
- The values in the New Entrant Information page is not comma-separated

Hi @patriciarussellCAS, I'm suggesting to move this back to In Progress to resolve the issues above. Please let me know if that works for you and @BCerki, thank you!

**Comment by BCerki** [comment]
> A few things I noticed:
> 
> * The `Equivalent Value` is not set to 4 decimal digits anymore, same with values in the Compliance Summary page, Emissions Summary page, Production Data page, and Additional Reporting Data page

I don't think they were ever meant to have 4 digits; they were meant to not be allowed more than 4 digits. Right now, they'll show as many digits as the user entered up to a max of 4. But quick to change if that's desired.
> * In the Allocation of Emissions page, the text fields are not aligned anymore

I think @ayeshmcg is changing the alignment as part of the PR that was blocked on this
> * The values in the New Entrant Information page is not comma-separated

I'll look into this. Update: there may be more forms that are missing the separator--I didn't realize reporting had some forms that only show up conditionally. I'll start a list here so we have it for testing too:
- new entrant (when registering an operation, select new entrant purpose)
- EIO  (when registering an operation, select EIO purpose)

**Comment by ayeshmcg** [comment]
@BCerki  The alignment issue mentioned by @fviduya  is UI issue , I think what is causing it is Total Emissions field should be like total allocated with no text box visible

<img width="1348" height="658" alt="Image" src="https://github.com/user-attachments/assets/7dd7e48a-87e5-443f-b66e-c9a87ac40c33" />

but now it shows a fields

<img width="1468" height="644" alt="Image" src="https://github.com/user-attachments/assets/241c5679-0c31-4d0b-b3dc-7556dd757ba6" />

This wil not be resolved by https://github.com/bcgov/cas-reporting/issues/820, as that is just to fix Equivalent Emissions in Report Raw Activity Data table

**Comment by patriciarussellCAS** [comment]
@BCerki there is also Reporting Operations (they have the same workflow as Regulated, minus the production data, allocation and compliance summary). 
Opt-ins should have the same workflow as Regulated. 
We also have a "simple report" however we haven't actually built that out yet so nothing to do there yet.

**Comment by BCerki** [comment]
> [@BCerki](https://github.com/BCerki) The alignment issue mentioned by [@fviduya](https://github.com/fviduya) is UI issue , I think what is causing it is Total Emissions field should be like total allocated with no text box visible
> 
> <img alt="Image" width="1348" height="658" src="https://private-user-images.githubusercontent.com/171871432/467111541-7dd7e48a-87e5-443f-b66e-c9a87ac40c33.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI2OTc2MDAsIm5iZiI6MTc1MjY5NzMwMCwicGF0aCI6Ii8xNzE4NzE0MzIvNDY3MTExNTQxLTdkZDdlNDhhLTg3ZTUtNDQzZi1iNjZlLWM5YTg3YWM0MGMzMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxNlQyMDIxNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZmZmY2Q0ODlhNzYyYzc5NzY2NmIyODkzNjQ0MGY3ZmVkMjM4YzFlNmZjNDZmYzFiNTM1ZGFiNzlmODAzZDE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4AlkIuiNxVv2ad8m9Wx6h0CFI-8j-Y8r4Jr_9SZtqZc">
> but now it shows a fields
> 
> <img alt="Image" width="1468" height="644" src="https://private-user-images.githubusercontent.com/171871432/467111696-241c5679-0c31-4d0b-b3dc-7556dd757ba6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI2OTc2MDAsIm5iZiI6MTc1MjY5NzMwMCwicGF0aCI6Ii8xNzE4NzE0MzIvNDY3MTExNjk2LTI0MWM1Njc5LTBjMzEtNGQwYi1iM2RjLTc1NTZkZDc1N2JhNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxNlQyMDIxNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iY2QwMzhiYzI1N2UzYzMyZjFlZDM4MDI3OWMyMTk2MmQzMzExODNmODBmZTZjMDVjMTdjYTAwYjgxMDdkMTI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.e16trd049EmHQ_Fa65iPb4G24m8HMsnJ92nFqtJEr1A">
> This wil not be resolved by [#820](https://github.com/bcgov/cas-reporting/issues/820), as that is just to fix Equivalent Emissions in Report Raw Activity Data table

Thanks! I was thinking of the alignment in the final review page, so unrelated here

**Comment by fviduya** [comment]
I have retested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EVgJJj4hhDlPoWyg9iA14-cBiMze1DPxkDxlLQx4QQ0Ulw?e=zgnTe8&nav=eyJoIjoiNzI2NzY0NTAxIn0%3D).

I can verify that:
- [x] I can verify that, in reporting
   - [x] When value > 999, it is comma-separated, tested in the following pages:
      - [x] Activity forms
      - [x] Emission Summary page
      - [x] Production Data page
      - [x] Allocation of Emissions page
      - [x] Additional Reporting Data page
      - [x] New Entrant Information page
      - [x] Operation Emission Summary
      - [x] Compliance Summary page
      - [ ] Final Review page - some parts are not comma-separated, but does not affect submission of report. Filed #842 for additional work
   - [x] When value <= 999, it is comma-separated, tested in all pages above
   - [x] The text fields in the Allocation of Emissions page are now aligned
   - [x] The decimal digits are consistent

Moving to Pending PO for review of test. Thank you!

**Comment by patriciarussellCAS** [comment]
Moving to accepted and noting the follow-up ticket. to review in TEST

**Comment by patriciarussellCAS** [comment]
Appears to be working in TEST.
